test/integration/kubernetes_deploy_test.rb

```diff
@@ -45,16 +45,39 @@ def test_service_account_predeployed_before_unmanaged_pod
       pod["spec"]["serviceAccountName"] = service_account_name
       pod["spec"]["automountServiceAccountToken"] = false
     end
-    # Expect the service account is deployed before the unmanaged pod
-    assert_deploy_success(result)
-    hello_cloud = FixtureSetAssertions::HelloCloud.new(@namespace)
-    hello_cloud.assert_configmap_data_present
-    hello_cloud.assert_all_service_accounts_up
-    hello_cloud.assert_unmanaged_pod_statuses("Succeeded", 1)
-    assert_logs_match_all([
-      %r{Successfully deployed in \d.\ds: ServiceAccount/build-robot},
-      %r{Successfully deployed in \d+.\ds: Pod/unmanaged-pod-.*},
-    ], in_order: true)
+
+    unless server_dry_run_available?
```

Comment:
Did you have to chnage this test because the unmanaged pod is actually invalid, but we never cared about the success/failure status/. If so let's just use a valid unmanged pod.

Reply:
sounds makes sense, invalid unmanaged pod could also make it fail, which is not something we would like to test here.

Reply:
Setting `PRINT_LOGS=1` will show you the logs (but don't rely on the pass/fail result because it causes the assert_log_match_all to be skipped). It appears what's happening is the server-dry-run is detecting that the service account does not yet exist and correctly failing: `error looking up service account k8sdeploy-test-unautomounted-service-account-p-e8e71a2e74b65512/build-robot: serviceaccount "build-robot" not found`. 

But this stinks because we know this can happen and are going to pre-deploy the service account so it will exist by the time the deployment goes out. I suspect this is going to be a problem for anything that can rely on something we pre-deploy.

To keep us from getting stuck 1 idea, but feel free to come up with something better, would be to just server-dry-run the items that are in the pre-deployed since they aren't supposed to be ordered / have dependencies.

Reply:
Think about this problem, server-dry-run only pre-deploy resource could solve the problem. 
But one idea come to my mind is separating pre-deploy and deploy resource at earlier stage. 
Before: initial deploy -> validate resource -> pre-deploy -> deploy all
Can we change to: initial deploy -> validate pre-deploy resource -> deploy pre-deploy -> validate other resource -> deploy other resource